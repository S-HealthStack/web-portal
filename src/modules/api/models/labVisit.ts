import { SQLTimestamp } from 'src/common/utils/datetime';

export type LabVisitItemResponse = {
  id: number;
  userId: string;
  startTime?: SQLTimestamp;
  endTime?: SQLTimestamp;
  checkedInBy?: string;
  notes?: string;
  filesPath?: string;
};

export type LabVisitListResponse = LabVisitItemResponse[];

export type LabVisitSaveItemRequest = Partial<LabVisitItemResponse>;

export type LabVisitParticipantSuggestionItemRequest = {
  limit: number;
};

export type LabVisitParticipantSuggestionItemResponse = { userId?: string };

export type LabVisitParticipantSuggestionResponse = {
  healthDataOverview?: LabVisitParticipantSuggestionItemResponse[];
};
