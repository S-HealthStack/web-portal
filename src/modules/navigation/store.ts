import { createSelector } from '@reduxjs/toolkit';
import { matchPath } from 'react-router-dom';
import { createBrowserHistory, createMemoryHistory, Location } from 'history';

import { RootState } from 'src/modules/store/store';

export enum Path {
  Root = '/',

  SignIn = '/signin',
  AccountActivation = '/account/activation',
  AccountCreate = '/account/create',
  AccountConfirm = '/account/confirm',
  AccountVerification = '/email-verification',
  ForgotPassword = '/account/forgot',
  ForgotPasswordConfirm = '/account/forgot/confirm',
  ResetPassword = '/account/forgot/reset',
  ResetPasswordComplete = '/account/forgot/complete',

  CreateStudy = '/create-study',

  Overview = '/overview',
  OverviewSubject = '/overview/participants/:subjectId',
  TrialManagement = '/trial-management',
  TrialManagementSubject = '/trial-management/participants/:subjectId',
  TrialManagementEditSurvey = '/trial-management/surveys/:surveyId/edit',
  TrialManagementSurveyResults = '/trial-management/surveys/:surveyId/results',
  UserAnalytics = '/user-analytics',
  DataCollection = '/data-collection',
  DataCollectionSubject = '/data-collection/participants/:subjectId',
  StudySettings = '/study-settings',
}

export const makeHistory = () =>
  process.env.NODE_ENV !== 'test'
    ? createBrowserHistory({ basename: process.env.PUBLIC_PATH })
    : createMemoryHistory();

export const history = makeHistory();

const sectionPaths = [
  Path.Overview,
  Path.OverviewSubject,
  Path.TrialManagement,
  Path.UserAnalytics,
  Path.DataCollection,
  Path.StudySettings,
] as const;

export const sectionPathSelector = createSelector(
  (state: RootState) => state.router,
  (r) => sectionPaths.find((s) => matchPath((r.location as unknown as Location).pathname, s))
);
