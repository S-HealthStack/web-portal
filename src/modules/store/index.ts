export type { RootState, AppDispatch, AppThunk } from './store';
export { useAppDispatch, useAppSelector } from './hooks';
export * from './tools';
