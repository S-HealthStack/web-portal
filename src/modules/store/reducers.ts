import { combineReducers } from '@reduxjs/toolkit';
import { connectRouter } from 'connected-react-router';
import type { Reducer } from 'redux';

import authReducer from 'src/modules/auth/auth.slice';
import studySettingsReducer from 'src/modules/study-settings/studySettings.slice';
import snackbarReducer from 'src/modules/snackbar/snackbar.slice';
import overviewReducers from 'src/modules/overview/overview.slice';
import studiesReducer from 'src/modules/studies/studies.slice';
import studyManagementReducer from 'src/modules/study-management/studyManagement.slice';
import dataCollectionReducer from 'src/modules/data-collection/dataCollection.slice';
import sidebarReducer from 'src/modules/main-layout/sidebar/sidebar.slice';
import fileUploadReducer from 'src/modules/file-upload/fileUpload.slice';
import { history } from 'src/modules/navigation/store';
import { DataSliceState } from 'src/modules/store/createDataSlice';
import forgotPasswordReducers from 'src/modules/auth/forgot-password/forgotPassword.slice';
import resetPasswordReducers from 'src/modules/auth/forgot-password/resetPassword.slice';
import objectStorageReducers from 'src/modules/object-storage/objectStorage.slice';

const createReducer = <T, V>(asyncReducers?: { [key: string]: Reducer<DataSliceState<T, V>> }) =>
  combineReducers({
    ...authReducer,
    router: connectRouter(history),
    snackbar: snackbarReducer,
    dataCollection: dataCollectionReducer,
    sidebar: sidebarReducer,
    ...studySettingsReducer,
    ...overviewReducers,
    ...studyManagementReducer,
    ...forgotPasswordReducers,
    ...resetPasswordReducers,
    ...fileUploadReducer,
    ...asyncReducers,
    ...objectStorageReducers,
    ...studiesReducer,
  });

export default createReducer;
