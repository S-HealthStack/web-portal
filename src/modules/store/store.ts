/* eslint-disable import/no-import-module-exports */
import {
  Action,
  ThunkAction,
  configureStore,
  getDefaultMiddleware,
  ThunkDispatch,
} from '@reduxjs/toolkit';
import { routerMiddleware } from 'connected-react-router';

import { history } from 'src/modules/navigation/store';
import rootReducer from 'src/modules/store/reducers';

export { rootReducer };

export const store = configureStore({
  reducer: rootReducer,
  middleware: [...getDefaultMiddleware(), routerMiddleware(history)],
  devTools: process.env.NODE_ENV !== 'production',
});

if (module.hot) {
  // Enable Webpack hot module replacement for reducers
  module.hot.accept('./reducers', async () => {
    // eslint-disable-next-line import/no-useless-path-segments
    const nextRootReducer = (await import('../store/reducers')).default;
    store.replaceReducer(nextRootReducer);
  });
}

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = ThunkDispatch<RootState, unknown, Action>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
