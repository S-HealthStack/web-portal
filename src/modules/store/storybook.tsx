import React from 'react';
import { configureStore, ConfigureStoreOptions } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { DecoratorFunction } from '@storybook/csf';
import { ReactFramework } from '@storybook/react';
import { MemoryRouter } from 'react-router-dom';
import { rootReducer, RootState } from './store';

export const getStoreDecorator = (state: ConfigureStoreOptions<RootState>['preloadedState']) => {
  const store = configureStore({
    reducer: rootReducer,
    preloadedState: state,
  });

  return ((story) => (
    <Provider store={store}>
      <MemoryRouter>{story()}</MemoryRouter>
    </Provider>
  )) as DecoratorFunction<ReactFramework>;
};
