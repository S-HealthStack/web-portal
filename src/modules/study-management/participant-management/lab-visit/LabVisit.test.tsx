import React from 'react';
import 'jest-styled-components';
import { act, render, screen, waitFor } from '@testing-library/react';
import theme from 'src/styles/theme';
import { makeStore } from 'src/modules/store/store';
import { history } from 'src/modules/navigation/store';
import { ThemeProvider } from 'styled-components';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import { fetchStudies } from 'src/modules/studies/studies.slice';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import LabVisit from './LabVisit';
import { labVisitSlice } from './labVisit.slice';

describe('LabVisit', () => {
  beforeAll(() => {
    Element.prototype.scrollTo = jest.fn();
    jest.setSystemTime(1680842869025);
  });

  it('should render', async () => {
    const store = makeStore();
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    await store.dispatch(fetchStudies({ force: true }));

    let container: HTMLElement | undefined;

    act(() => {
      container = render(
        <ThemeProvider theme={theme}>
          <Provider store={store}>
            <ConnectedRouter history={history}>
              <LabVisit />
            </ConnectedRouter>
          </Provider>
        </ThemeProvider>
      ).container;
    });

    await waitFor(() => !!labVisitSlice.stateSelector(store.getState()).data);

    expect(container?.firstChild).toMatchSnapshot();

    // notes modal
    const firstShowNotesBtn = (await screen.findAllByTestId('show-notes-button'))[0];
    expect(firstShowNotesBtn).toBeInTheDocument();
    await userEvent.click(firstShowNotesBtn);

    const notesEditorTitle = await screen.findByTestId('modal-title');
    expect(notesEditorTitle).toBeInTheDocument();
    expect(notesEditorTitle.textContent).toEqual('Notes');

    const declineNotesEditorBtn = await screen.findByTestId('decline-button');
    expect(declineNotesEditorBtn).toBeInTheDocument();
    await userEvent.click(declineNotesEditorBtn);

    await userEvent.click(firstShowNotesBtn);
    const editNotesEditorBtn = await screen.findByTestId('accept-button');
    expect(editNotesEditorBtn).toBeInTheDocument();
    await userEvent.click(editNotesEditorBtn);

    // edit modal
    const firstEditVisitBtn = (await screen.findAllByTestId('edit-visit-button'))[0];
    expect(firstEditVisitBtn).toBeInTheDocument();
    await userEvent.click(firstEditVisitBtn);

    const visitEditorTitle = await screen.findByTestId('modal-title');
    expect(visitEditorTitle).toBeInTheDocument();
    expect(visitEditorTitle.textContent).toEqual('Edit In-Lab Visit');

    const declineVisitEditorBtn = await screen.findByTestId('decline-button');
    expect(declineVisitEditorBtn).toBeInTheDocument();
    await userEvent.click(declineVisitEditorBtn);

    // add modal
    const addVisitBtn = await screen.findByTestId('add-visit-button');
    expect(addVisitBtn).toBeInTheDocument();
    await userEvent.click(addVisitBtn);
  });

  it('[NEGATIVE] should render with API error', async () => {
    const store = makeStore();
    act(() => {
      render(
        <ThemeProvider theme={theme}>
          <Provider store={store}>
            <ConnectedRouter history={history}>
              <LabVisit />
            </ConnectedRouter>
          </Provider>
        </ThemeProvider>
      );
    });

    await waitFor(() => !!labVisitSlice.stateSelector(store.getState()).error);

    expect(screen.getByText('No Data')).toBeInTheDocument();
  });
});
