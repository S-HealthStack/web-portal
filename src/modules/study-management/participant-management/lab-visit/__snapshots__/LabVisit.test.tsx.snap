// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabVisit should render 1`] = `
.c10 {
  position: absolute;
  will-change: opacity,transform;
  border-radius: 50%;
  -webkit-animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  z-index: 0;
  background-color: #E3EAFB;
  pointer-events: none;
}

.c7 {
  position: relative;
  z-index: 0;
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 0.25rem;
  width: 100%;
  white-space: nowrap;
  padding: 0;
  overflow: hidden;
  -webkit-transition-duration: 400ms;
  transition-duration: 400ms;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  height: 2.5rem;
  box-sizing: border-box;
  border: 0.0625rem solid #4475E3;
  background-color: transparent;
}

.c7:hover {
  cursor: pointer;
}

.c7 > div {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 600;
  font-style: normal;
}

.c7 > div svg {
  margin-right: 0.25rem;
}

.c7 > div {
  height: 100%;
  color: #4475E3;
}

.c7 > div svg {
  fill: #4475E3;
}

.c7:hover:enabled {
  background-color: #ECF1FC;
}

.c7:disabled {
  border: 0.0625rem solid #B3C6F1;
}

.c7:disabled > div {
  color: #B3C6F1;
}

.c7:disabled > div svg {
  fill: #B3C6F1;
}

.c7:focus-visible {
  outline: 0.125rem solid #C7D6F7;
}

.c42 {
  position: relative;
  z-index: 0;
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 0.25rem;
  width: 100%;
  white-space: nowrap;
  padding: 0;
  overflow: hidden;
  -webkit-transition-duration: 0;
  transition-duration: 0;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  height: 1.5rem;
  width: 1.5rem;
  border: none;
  background-color: transparent;
}

.c42 > div {
  color: #4475E3;
}

.c42 > div svg {
  fill: #4475E3;
}

.c42:hover:enabled > div {
  color: #2F65E0;
}

.c42:hover:enabled > div svg {
  fill: #2F65E0;
}

.c42:hover:enabled:active > div {
  color: #2057D5;
}

.c42:hover:enabled:active > div svg {
  fill: #2057D5;
}

.c42:disabled > div {
  color: #B3C6F1;
}

.c42:disabled > div svg {
  fill: #B3C6F1;
}

.c42:focus-visible {
  outline: 0.125rem solid #C7D6F7;
}

.c42:focus-visible > div {
  color: #4475E3;
}

.c42:focus-visible > div svg {
  fill: #4475E3;
}

.c42:hover:enabled {
  background-color: #ECF1FC;
}

.c42:hover:enabled:active > div {
  color: #2057D5;
}

.c42:hover:enabled:active > div svg {
  fill: #2057D5;
}

.c42:focus-visible:enabled {
  background-color: #C7D6F7;
  outline: none;
}

.c9 {
  position: relative;
  z-index: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transform: translate3d(0,0,0.0625rem);
  -ms-transform: translate3d(0,0,0.0625rem);
  transform: translate3d(0,0,0.0625rem);
}

.c20 {
  display: inline-block;
  position: relative;
}

.c35 {
  position: relative;
  width: 100%;
}

.c40 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
  grid-area: label;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.c41 {
  grid-area: right-icon;
}

.c38 {
  width: 100%;
  height: 1.5625rem;
  padding: 0 1rem;
  padding-right: 0.5rem;
  box-sizing: border-box;
  display: grid;
  grid-template-areas: ' label right-icon';
  grid-template-columns: 1fr 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.25rem;
  background-color: #FFFFFF;
  border: 0.0625rem solid transparent;
  border-top-width: 0;
  -webkit-transition: all 300ms cubic-bezier(0.45,0.05,0,1);
  transition: all 300ms cubic-bezier(0.45,0.05,0,1);
}

.c38:hover {
  cursor: pointer;
}

.c39 {
  border-top-width: 0.0625rem;
  border-color: !important;
  border-radius: 0.25rem;
}

.c39:hover {
  border-color: #2F65E0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #FFFFFF;
  box-shadow: 0 0 0.125rem 0rem rgba(0,0,0,0.15);
  border-radius: 0.25rem;
  box-sizing: border-box;
  padding: 1.5rem 1.5rem 1rem;
  overflow: visible;
  position: relative;
  pointer-events: none;
}

.c0 * {
  cursor: default;
}

.c0 > * {
  min-width: 0;
  min-height: 0;
  overflow: visible;
}

.c11 {
  opacity: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 2.6875rem;
}

.c4 {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.3125rem;
  font-weight: 600;
  font-style: normal;
  color: #474747;
  display: inline-block;
  margin: 0;
}

.c5 {
  margin-left: 0.5625rem;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-animation: AGEkE 1.25s;
  animation: AGEkE 1.25s;
}

.c6 {
  display: none;
  overflow: visible;
  position: relative;
}

.c29 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  line-height: 1.75rem;
}

.c31 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #4475E3;
  line-height: 1.75rem;
}

.c32 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-right: -0.125rem;
}

.c43 {
  margin-left: 1rem;
}

.c43:disabled > div {
  background-color: transparent;
}

.c43:disabled > div svg {
  fill: #DBDBDB;
}

.c33 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 2.5625rem;
}

.c34 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #4475E3;
  margin-right: 0.5rem;
}

.c36 {
  width: 2.5rem;
  height: 1.5625rem;
  border: none;
  border-radius: 0.25rem;
  z-index: 10;
  color: #808080;
}

.c36 > .c37 {
  position: relative;
  z-index: 100;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-left: 0.375rem;
  padding-right: 0.375rem;
  border-color: #DBDBDB;
}

.c36 > .c37:hover {
  cursor: pointer;
}

.c36 > .c37 > svg {
  display: none;
}

.c22 {
  background: none;
  border: none;
  padding: 0;
  height: 1.5rem;
  width: 1.5rem;
}

.c22 svg {
  fill: #DBDBDB;
}

.c22:active,
.c22:enabled:hover {
  cursor: pointer;
}

.c22:active svg,
.c22:enabled:hover svg {
  fill: #DBDBDB;
}

.c22:disabled svg {
  fill: #DBDBDB;
}

.c18 {
  position: relative;
  box-sizing: border-box;
  height: 2.125rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.4375rem 0.5rem;
  color: #2D2D2D;
  overflow: hidden;
  -webkit-transition: background-color 0.3s cubic-bezier(0.45,0.05,0,1);
  transition: background-color 0.3s cubic-bezier(0.45,0.05,0,1);
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c18 > span {
  font-family: Inter,sans-serif;
  font-size: 0.75rem;
  line-height: 1.125rem;
  font-weight: 400;
  font-style: normal;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  min-width: 100%;
}

.c21 {
  width: 1.5rem;
  height: 1.5rem;
  margin-left: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c19 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #4475E3;
  box-shadow: inset 0 -0.0625rem 0 #ECF1FC;
  background-color: #FFFFFF;
}

.c19 > * {
  -webkit-transition: background 0.3s cubic-bezier(0.45,0.05,0,1);
  transition: background 0.3s cubic-bezier(0.45,0.05,0,1);
}

.c19:hover {
  background-color: #F7F8FA;
  cursor: pointer;
}

.c19 > span {
  min-width: auto;
  font-family: Inter,sans-serif;
  font-size: 0.75rem;
  line-height: 1.125rem;
  font-weight: 600;
  font-style: normal;
}

.c19 > span:first-letter {
  text-transform: uppercase;
}

.c23 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #4475E3;
  box-shadow: inset 0 -0.0625rem 0 #ECF1FC;
  background-color: #FFFFFF;
}

.c23 > * {
  -webkit-transition: background 0.3s cubic-bezier(0.45,0.05,0,1);
  transition: background 0.3s cubic-bezier(0.45,0.05,0,1);
}

.c23:hover {
  background-color: #F7F8FA;
}

.c23 > span {
  min-width: auto;
  font-family: Inter,sans-serif;
  font-size: 0.75rem;
  line-height: 1.125rem;
  font-weight: 600;
  font-style: normal;
}

.c23 > span:first-letter {
  text-transform: uppercase;
}

.c16 {
  position: relative;
  display: grid;
  max-height: 2.125rem;
  min-height: 2.125rem;
  -webkit-column-gap: 1.5rem;
  column-gap: 1.5rem;
  box-shadow: inset 0 -0.0625rem 0 #ECF1FC;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  height: 27.4375rem !important;
}

.c14 {
  -webkit-scrollbar-width: thin;
  -moz-scrollbar-width: thin;
  -ms-scrollbar-width: thin;
  scrollbar-width: thin;
  -webkit-scrollbar-color: #4475E3 #FFFFFF;
  -moz-scrollbar-color: #4475E3 #FFFFFF;
  -ms-scrollbar-color: #4475E3 #FFFFFF;
  scrollbar-color: #4475E3 #FFFFFF;
}

.c14::-webkit-scrollbar {
  width: 0.25rem;
  height: 0.25rem;
}

.c14::-webkit-scrollbar-track {
  background: #FFFFFF;
  border-radius: 0.25rem;
  margin: 0.0625rem 0;
}

.c14::-webkit-scrollbar-thumb {
  background-color: #4475E3;
  border-radius: 0.625rem;
  border: 0rem solid #FFFFFF;
  width: 0.25rem;
  min-height: 0.25rem;
}

.c15 {
  position: relative;
  overflow: auto;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  width: 100%;
  height: 100%;
}

.c17 {
  background-color: #FFFFFF;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  z-index: 3;
}

.c27 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  width: 100%;
  position: relative;
  min-height: calc(100% - 2.164375rem);
}

.c28 {
  margin-top: 1.25rem;
  height: 2rem;
}

.c30 {
  margin-left: 0.125rem;
}

.c24 {
  pointer-events: none;
  position: absolute;
  top: 2.164375rem;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c25 {
  display: grid;
  -webkit-column-gap: 1.5rem;
  column-gap: 1.5rem;
}

.c26 {
  -webkit-transition: background-color 0.3s cubic-bezier(0.45,0.05,0,1);
  transition: background-color 0.3s cubic-bezier(0.45,0.05,0,1);
  background: transparent;
  height: 100%;
}

.c1 {
  min-height: 33rem;
}

.c8 {
  margin-top: -0.5rem;
  width: 10.25rem;
}

.c13 {
  margin-top: 0.625rem;
}

<div
  class="c0 c1"
>
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <p
        class="c4"
        data-testid="title"
      >
        In-lab Visit
      </p>
      <svg
        class="c5"
        data-testid="loader"
      />
      <div
        class="c6"
      >
        <button
          class="c7 c8"
          data-testid="add-visit-button"
          fill="bordered"
          style="overflow: hidden; position: relative;"
        >
          <div
            class="c9"
            data-testid="content"
          >
            <svg />
            Add in-lab visit
          </div>
          <div
            class="c10"
            color="primaryLightPressed"
            opacity="1,1"
            scale="0"
            style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    class="c11"
    data-testid="card-content"
  >
    <div
      class="c12 c13"
      data-testid="table"
    >
      <div
        class=" c14 c15"
        style="webkit-overflow-scrolling: touch;"
      >
        <div
          class="c16 c17"
          style="grid-template-columns: 159px 159px 159px 159px 83px 116px 53px;"
        >
          <div
            class="c18 c19"
            data-testid="table-head-cell"
          >
            <span
              class="c20"
              data-testid="tooltip-container"
            >
              Participant ID
            </span>
            <div
              class="c21"
            >
              <button
                aria-label="asc"
                class="c22"
                color="disabled"
              >
                <svg
                  data-testid="icon"
                />
              </button>
            </div>
          </div>
          <div
            class="c18 c19"
            data-testid="table-head-cell"
          >
            <span
              class="c20"
              data-testid="tooltip-container"
            >
              Visit Start
            </span>
            <div
              class="c21"
            >
              <button
                aria-label="Sort disabled"
                class="c22"
                color="disabled"
              >
                <svg
                  data-testid="icon"
                />
              </button>
            </div>
          </div>
          <div
            class="c18 c19"
            data-testid="table-head-cell"
          >
            <span
              class="c20"
              data-testid="tooltip-container"
            >
              Visit End
            </span>
            <div
              class="c21"
            >
              <button
                aria-label="Sort disabled"
                class="c22"
                color="disabled"
              >
                <svg
                  data-testid="icon"
                />
              </button>
            </div>
          </div>
          <div
            class="c18 c19"
            data-testid="table-head-cell"
          >
            <span
              class="c20"
              data-testid="tooltip-container"
            >
              Check-in by
            </span>
            <div
              class="c21"
            >
              <button
                aria-label="Sort disabled"
                class="c22"
                color="disabled"
              >
                <svg
                  data-testid="icon"
                />
              </button>
            </div>
          </div>
          <div
            class="c18 c19"
            data-testid="table-head-cell"
          >
            <span
              class="c20"
              data-testid="tooltip-container"
            >
              Notes
            </span>
            <div
              class="c21"
            >
              <button
                aria-label="Sort disabled"
                class="c22"
                color="disabled"
              >
                <svg
                  data-testid="icon"
                />
              </button>
            </div>
          </div>
          <div
            class="c18 c19"
            data-testid="table-head-cell"
          >
            <span
              class="c20"
              data-testid="tooltip-container"
            >
              Documents
            </span>
            <div
              class="c21"
            >
              <button
                aria-label="Sort disabled"
                class="c22"
                color="disabled"
              >
                <svg
                  data-testid="icon"
                />
              </button>
            </div>
          </div>
          <div
            class="c18 c23"
            data-testid="table-head-cell"
          >
            <span
              class="c20"
              data-testid="tooltip-container"
            />
          </div>
        </div>
        <div
          class="c24"
          style="height: 0rem;"
        >
          <div
            class="c25"
            style="grid-template-columns: 159px 159px 159px 159px 83px 116px 53px;"
          >
            <div
              class="c26"
            />
            <div
              class="c26"
            />
            <div
              class="c26"
            />
            <div
              class="c26"
            />
            <div
              class="c26"
            />
            <div
              class="c26"
            />
            <div
              class="c26"
            />
          </div>
        </div>
        <div
          class="c27"
          style="pointer-events: auto;"
        />
      </div>
      <div
        class="c28"
      >
        <div
          class="c29 c30"
          data-testid="pagination"
        >
          <div
            class="c31 c30"
            data-testid="info"
          >
            1-0 of 0
          </div>
          <div
            class="c32 c30"
          >
            <div
              class="c33"
            >
              <span
                class="c34"
              >
                Rows per page
              </span>
              <div
                class="c35 c36"
              >
                <div
                  class="c37 c38 c39"
                  data-testid="value-item"
                >
                  <div
                    class="c40"
                  >
                    10
                  </div>
                  <svg
                    class="c41"
                  />
                </div>
              </div>
            </div>
            <button
              aria-label="Go to First"
              class="c42 c43"
              data-testid="go-to-first"
              disabled=""
              fill="text"
              style="overflow: hidden; position: relative;"
            >
              <div
                class="c9"
                data-testid="content"
              >
                <svg />
              </div>
              <div
                class="c10"
                color="primaryLightPressed"
                opacity="1,1"
                scale="0"
                style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
              />
            </button>
            <button
              aria-label="Go to Previous"
              class="c42 c43"
              data-testid="go-to-previous"
              disabled=""
              fill="text"
              style="overflow: hidden; position: relative;"
            >
              <div
                class="c9"
                data-testid="content"
              >
                <svg />
              </div>
              <div
                class="c10"
                color="primaryLightPressed"
                opacity="1,1"
                scale="0"
                style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
              />
            </button>
            <button
              aria-label="Go to Next"
              class="c42 c43"
              data-testid="go-to-next"
              disabled=""
              fill="text"
              style="overflow: hidden; position: relative;"
            >
              <div
                class="c9"
                data-testid="content"
              >
                <svg />
              </div>
              <div
                class="c10"
                color="primaryLightPressed"
                opacity="1,1"
                scale="0"
                style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
              />
            </button>
            <button
              aria-label="Go to Last"
              class="c42 c43"
              data-testid="go-to-last"
              disabled=""
              fill="text"
              style="overflow: hidden; position: relative;"
            >
              <div
                class="c9"
                data-testid="content"
              >
                <svg />
              </div>
              <div
                class="c10"
                color="primaryLightPressed"
                opacity="1,1"
                scale="0"
                style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
